test/src/test/java/hudson/model/JobTest.java

```diff
@@ -210,6 +211,9 @@ public String getDisplayName() {
 
     @LocalData
     @Test public void configDotXmlPermission() throws Exception {
+        // legacy behavior re-enabled (could be changed when the webclient will be adapted
+        ApiTokenPropertyConfiguration.get().setTokenGenerationOnCreationEnabled(true);
```

Comment:
NIT: could be better as a test helper method that is called, so that when the TODO is done it will be easier to change all the test code.